packages/itwinui-react/src/core/InputGrid/InputGrid.tsx

```diff
@@ -16,6 +16,10 @@ type InputGridOwnProps = {
    * @default 'default'
    */
   labelPlacement?: 'default' | 'inline';
+  /**
+   * Status of the InputGrid (and its descendants).
+   */
+  status?: 'positive' | 'warning' | 'negative';
```

Comment:
Added the `status` prop since in itwinui-css, the input grid needs `data-iui-status` to correctly add status colored outlines.

Reply:
how was it working previously then? we weren't using the CSS that sets `data-iui-status`?

Reply:
The outline was previously coming from the input-flex-container:

![image](https://github.com/iTwin/iTwinUI/assets/45748283/0ce2c669-7f48-490b-ba89-a10c4c6801de)

… and `--iui-color-border-negative` was being by the `data-iui-status` on the input-flex-container

---

Now, it is coming from input-with-icon:

![Screenshot 2024-01-12 at 10 24 37 AM](https://github.com/iTwin/iTwinUI/assets/45748283/25bdcf7e-c13b-404d-be0b-490b1235be53)

… and `--_iui-input-with-icon-focus-color` was being set by the `data-iui-status` on the input grid

![image](https://github.com/iTwin/iTwinUI/assets/45748283/c93f78df-1c2c-461d-82c6-c123e6df2584)

https://github.com/iTwin/iTwinUI/blob/5d8986451e10101085c98a084fcc4a67b0d76d1e/packages/itwinui-css/src/input-container/input-container.scss#L66

Reply:
So we weren't using `iui-input-grid[data-iui-status]` anywhere in React before, correct?

Since `InputGrid` is a public component and is exposing a public `status` prop, I would expect it to work with various permutations, including:
- `Input` within `InputGrid`
- `InputWithDecorations` within `InputGrid`

If such combinations require more work, then we should get rid of the `status` prop and manually set `data-iui-status` inside `LabeledInput` (for now)

Reply:
> So we weren't using `iui-input-grid[data-iui-status]` anywhere in React before, correct?

It is being used by our private `InputContainer` component. 

https://github.com/iTwin/iTwinUI/blob/5d8986451e10101085c98a084fcc4a67b0d76d1e/packages/itwinui-react/src/core/utils/components/InputContainer.tsx#L48-L50

`InputContainer` is used only in `ComboBoxInputContainer`.

---

Actually, after looking at `InputContainer`, I realized that maybe I can re-use `InputContainer` inside `LabeledInput`, similar to how `ComboBoxInputContainer` does it. If so, they we can skip adding the new `status` to `InputGrid`.

Reply:
no, I think we should remove `InputContainer` to make things less confusing

Reply:
see relevant comment: https://github.com/iTwin/iTwinUI/pull/1355#discussion_r1277577290

Reply:
Then since this PR has already handles many cases and since this is a pre-requisite for #1777, I think it would be better to add a `status` prop to `InputGrid` in a future PR.

So, I will delete the `status` prop and add `data-iui-status` manually. I will also create a checklist item in the "TODOs after PR" section of this PR to consider adding `status` to `InputGrid` and try removing `InputContainer` in a future PR.

Reply:
Done in [`405c5cf` (#1753)](https://github.com/iTwin/iTwinUI/pull/1753/commits/405c5cff60877e34fa93b118a074a105cba158ba)